patches/tModLoader/Terraria/ID/TileID.TML.cs

```diff
@@ -16,6 +16,28 @@ partial class Sets
 		/// <summary> Allows non-solid tiles to be sloped (solid tiles can always be sloped, regardless of this set). </summary>
 		public static bool[] CanBeSloped = Factory.CreateBoolSet();
 
+		/// <summary>
+		/// The ID of the projectile for this falling block. Defaults to -1.
+		/// </summary>
+		public static int[] FallingBlockProjectileID = Factory.CreateIntSet(-1,
+			Sand, ProjectileID.SandBallFalling,
+			Ebonsand, ProjectileID.EbonsandBallFalling,
+			TileID.Mud, ProjectileID.MudBall,
```

Comment:
Any reason you qualify `Mud` here (and farther below).

Reply:
It needs it cause there's a TileID.Sets.Mud as well (possibly used for the jungle?)

Reply:
Mud Blocks were affected by gravity in 1.0.5, then it was reverted in 1.0.6 (same with Ash) https://terraria.wiki.gg/wiki/Mud_Block#History
What tomat is talking about is how you put `TileID.Mud` instead of just `Mud`.

Reply:
Yeah, in TileID there's a `TileID.Sets.Mud`, which if I use just `Mud`, it gives an error, since `TileID.Sets.Mud` is a bool array and can't be used as an `int`. The exact error is "Argument 6: cannot convert from 'bool[]' to 'int'".